### PR TITLE
Topic3

### PR DESCRIPTION
**Pull request by aguirre-luciano**
Topic 3 Key Points 1 3 and 4

### CODE REVIEW

File: KP1-Topic3/src/com/aguirre/tests/ConvertTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		String text = conv.convert(1000.00);
+		assertEquals("one thousand dollars",text);
+		text = conv.convert(6552.05);
+		assertEquals("six thousand five hundred fifty-two and 05/100 dollars",text);
+		text = conv.convert(301999.86);
+		assertEquals("three hundred one thousand nine hundred ninety-nine and 86/100 dollars",text);
+		text = conv.convert(91386.20);
+		assertEquals("ninety-one thousand three hundred eighty-six and 20/100 dollars",text);
+	}
+}
```
Comment (amilcar-infante) [inline KP1-Topic3/src/com/aguirre/tests/ConvertTest.java:75]: Well done!!!

File: KP1-Topic3/src/com/aguirre/classes/Converter.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		}
+
+		// if the amount is between 1.00 dollars and 999.99 dollars
+		if (integerPart <= 999) {
+			text = convertUntil999(integerPart) + " " + text;
+			return text;
+		}
+
+		// if the amount is bigger than 999999.99
+		if (integerPart >= 1000000) {
```
Comment (amilcar-infante) [inline KP1-Topic3/src/com/aguirre/classes/Converter.java:86]: This condition wasn't validated in the tests

File: KP1-Topic3/src/com/aguirre/classes/Converter.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+		tensNames[4] = "forty";
+		tensNames[5] = "fifty";
+		tensNames[6] = "sixty";
+		tensNames[7] = "seventy";
+		tensNames[8] = "eighty";
+		tensNames[9] = "ninety";
+
+		bigNumbers = new String[2];
+		bigNumbers[0] = "thousand";
+		bigNumbers[1] = "million";
```
Comment (amilcar-infante) [inline KP1-Topic3/src/com/aguirre/classes/Converter.java:48]: With the condition declared on line 86 isn't needed bigNumber "million"

File: KP3-Topic3/src/com/aguirre/test/RecentFileListTest.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+		assertTrue("The list should be full", list.fullList());
+		assertTrue("The oldest file should be File 1", list.oldestFile().equals(f1));
+		File f16 = new File("F16");
+		list.addFile(f16);
+		assertTrue("The list should be full", list.fullList());
+		assertTrue("The oldest file should be File 2", list.oldestFile().equals(f2));
+		assertTrue("The most recent file is incorrect", list.mostRecentFile().equals(f16));
+	}
+}
```
Comment (amilcar-infante) [inline KP3-Topic3/src/com/aguirre/test/RecentFileListTest.java:130]: Good job!!! and excellent tests!! :+1
I would add a test similar to fullList's testcase but adding at the end one element that is in the middle of the list to validate that it is also working with the list full of elements

File: KP4-Topic3/src/com/aguirre/classes/RomanNumberConverter.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+	public int converToInt(String number) {
+		int decimal = 0;
+		int lastDigit = 0;
+		int currentDigit = 0;
+		int digit;
+		for (int i = 0; i < number.length(); i++) {
+			digit = number.charAt(i);
+			switch (digit) {
+			case 'I': {
```
Comment (amilcar-infante) [inline KP4-Topic3/src/com/aguirre/classes/RomanNumberConverter.java:28]: Case should be indented

File: KP4-Topic3/src/com/aguirre/classes/RomanNumberConverter.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+				currentDigit = 500;
+				break;
+			}
+			case 'M': {
+				currentDigit = 1000;
+				break;
+			}
+			}
+
+			// If lastDigit is smaller than currentDigit, we subtract
```
Comment (amilcar-infante) [inline KP4-Topic3/src/com/aguirre/classes/RomanNumberConverter.java:58]: It's an excellent practice add comments in the code :+1:

File: KP4-Topic3/src/com/aguirre/tests/RomanNumberConverterTest.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+import com.aguirre.classes.RomanNumberConverter;
+
+public class RomanNumberConverterTest {
+
+	@Test
+	public void oneLetter() {
```
Comment (amilcar-infante) [inline KP4-Topic3/src/com/aguirre/tests/RomanNumberConverterTest.java:12]: It's a good practice to try to explain in the name of the testcase exactly what are you testing, for example in this one you could use testRomanToIntWithOneLetter

File: KP4-Topic3/src/com/aguirre/tests/RomanNumberConverterTest.java [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+		String num = converter.converToRomanNumber(156);
+		assertEquals("CLVI", num);
+		num = converter.converToRomanNumber(2015);
+		assertEquals("MMXV", num);
+		num = converter.converToRomanNumber(777);
+		assertEquals("DCCLXXVII", num);
+		num = converter.converToRomanNumber(1444);
+		assertEquals("MCDXLIV", num);
+	}
+}
```
Comment (amilcar-infante) [inline KP4-Topic3/src/com/aguirre/tests/RomanNumberConverterTest.java:127]: Great job!!! From my point of view you covered all the RomanNumberConverter class :+1:

**Comment by amilcar-infante** [comment]
Good job Luciano with the points 1, 3 and 4 of this topic!!! :+1: 
I will be waiting the keypoint 2 to do the review next Monday
Excellent work with git, you uploaded only the required files :+1: :+1: :+1:

File: KP1-Topic3/src/com/aguirre/classes/Converter.java (hunk 2 above)
Comment (aguirre-luciano) [inline KP1-Topic3/src/com/aguirre/classes/Converter.java:86]: Yes, sorry!

File: KP1-Topic3/src/com/aguirre/classes/Converter.java (hunk 3 above)
Comment (aguirre-luciano) [inline KP1-Topic3/src/com/aguirre/classes/Converter.java:48]: Yes, I forgot to delete it

File: KP3-Topic3/src/com/aguirre/test/RecentFileListTest.java (hunk 4 above)
Comment (aguirre-luciano) [inline KP3-Topic3/src/com/aguirre/test/RecentFileListTest.java:130]: Yes, you are right (I tested recently and also works)

File: KP4-Topic3/src/com/aguirre/classes/RomanNumberConverter.java (hunk 5 above)
Comment (aguirre-luciano) [inline KP4-Topic3/src/com/aguirre/classes/RomanNumberConverter.java:28]: Yes, my mistake

File: KP4-Topic3/src/com/aguirre/tests/RomanNumberConverterTest.java (hunk 7 above)
Comment (aguirre-luciano) [inline KP4-Topic3/src/com/aguirre/tests/RomanNumberConverterTest.java:12]: Thanks, I like to do it but sometimes the name is too long

Comment (amilcar-infante) [inline KP4-Topic3/src/com/aguirre/tests/RomanNumberConverterTest.java:12]: From my point of view there is no problem with the length of test method, but as I said "from my point of view"

**Comment by amilcar-infante** [comment]
Good job, this topic is closed

**Pull request closed without merging**